certora/specs/SafeWebAuthnSignerFactory.spec

```diff
@@ -115,3 +123,41 @@ rule getSignerRevertingConditions {
 
     assert lastReverted <=> triedTransferringEth;
 }
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Correctness of Signer Creation. (Proved)                                                                            │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+
+ghost mathint numOfCreation;
+ghost mapping(address => uint) address_map;
+ghost address signerAddress;
+
+hook EXTCODESIZE(address addr) uint v{
+    require address_map[addr] == v;
+}
+
+hook CREATE2(uint value, uint offset, uint length, bytes32 salt) address v{
+    require(v == signerAddress);
+    numOfCreation = numOfCreation + 1;
+    address_map[v] = length;
+}
+
+rule SignerCreationCantOverride()
+{
+    env e;
+    require numOfCreation == 0;
+
+    uint x;
+    uint y;
+    P256.Verifiers verifier;
+
+    address a = getSigner(e, x, y, verifier);
+    require address_map[a] == 0;
+
+    createSigner(e, x, y, verifier);
+    createSigner@withrevert(e, x, y, verifier);
```

Comment:
does it make sense to add f(e,args) between the two calls to createSigner, to show that also after any function you can still not override ?

Reply:
It was done (except for proxy fallback which is causing havoc on the delegate call.